### ISSUE TITLE
parse_event_source still mis-identifies some sources

### ISSUE THREAD

**Issue by juzna** [DataDog/datadog-serverless-functions#344]
https://github.com/DataDog/datadog-serverless-functions/pull/299 tried to fix this, but it's still happens and we've just encountered this.

**Describe what happened:**
We have a S3 bucket named `cool-project-lambda-bucket`, which is used by a lambda in a cool project. This bucket has *Server access logging* enabled to `s3-log-bucket` with a using the original bucket name (i.e. `cool-project-lambda-bucket/*`). Logs from the logging bucket are read by this log-shipping lambda and sent to Datadog.

The log-shipping lambda is triggered for a new log file with key e.g. `cool-project-lambda-bucket/123.log`. 

`parse_event_source` determines that the source is `lambda` instead of `s3`.

**Describe what you expected:** `parse_event_source` should recognise correctly that the source is `s3`.

**Steps to reproduce the issue:**
1. create an S3 bucket with `lambda` in its name
2. enable Server access logging to a log bucket, use the first bucket name as a file prefix
3. get logs shipped to Datadog; they won't be parsed as S3 logs

**Failing Test Case:**
```py
# tests/test_parse_event_source.py
    def test_s3_source_path_with_keyword(self):
        self.assertEqual(parse_event_source({"Records": ["logs-from-s3"]}, "cool-lambda/2020-09-09-07-40-22-ABC12345"), "s3")
```

**Comment by tianchu** [comment]
cc: @DylanLovesCoffee, can you update this issue when a fix is available?

**Comment by agocs** [comment]
I believe this is fixed in #365
